exercise-solutions/iterators/src/main.rs

```diff
@@ -0,0 +1,22 @@
+use std::error::Error;
+use std::fs::File;
+use std::io::{BufRead, BufReader};
+
+fn main() -> Result<(), Box<dyn Error>> {
+    let f = File::open("numbers.txt")?;
+    let reader = BufReader::new(f);
+
+    let sum_of_odd_numbers: i32 = reader
+        .lines()
+        .filter_map(|line| line.ok())
+        .filter_map(|s| s.parse().ok())
+        .filter(|num| num % 2 != 0)
+        .fold(0, |acc, elem| acc + elem);
+        // Also works
+        //.sum::<i32>();
```

Comment:
I would simply put `.sum();` here. Unlike `filter` and `map` that are popular, `fold` is not used too often. Half of `Itertools` methods are fold variants with more descriptive methods. May as well call `sum` because we work with numbers here. Turbofish is not needed, because there's `i32` on line 9 already.

```suggestion
        .sum();
```

Reply:
```text
11 |     let result = reader.lines()
   |         ^^^^^^
...
15 |         .sum();
   |          --- type must be known at this point
```

oops